File: piper/extensions/chrome/src/engine/statements/lists_create_with.js
import { stringify } from "../util"


export default async function(statement, onError) {
    try {
        let pointer = statement.pointer
        let output = []
        for (let item of pointer) {
            let value = await this.executeStatement(stringify(item))
            output.push(value)
        }
        return output
    } catch (e) {
        onError(e)
        return []
    }
}

File: piper/android/app/src/main/java/io/gopiper/piper/engine/statements/lists_create_with.kt
package io.gopiper.piper.engine.statements

import android.util.Log
import io.gopiper.piper.engine.Piper
import org.json.JSONObject
import java.lang.Exception

suspend fun Piper.executeListsCreateWith(statement: JSONObject, onError: (e: Exception) -> Unit): ArrayList<Any?> {
    try {
        Log.d("hello", "execute_lists_create_with")
        val pointer = statement.getJSONArray("pointer")
        val output = ArrayList<Any?>()
        for (i in 0 until pointer.length()) {
            val item = executeStatement(pointer[i].toString())
            output.add(item)
        }
        return output
    } catch (e: Exception) {
        onError(e)
        return ArrayList()
    }
}